mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MessageToElasticSearchJson.java

```diff
@@ -85,9 +97,21 @@ public String convertToJsonWithoutAttachment(MailboxMessage message, List<User>
         Preconditions.checkNotNull(message);
         switch (indexMessageId) {
             case Required:
-                return mapper.writeValueAsString(IndexableMessageWithMessageId.from(message, users, textExtractor, zoneId, IndexAttachments.NO));
+                return mapper.writeValueAsString(IndexableMessage.builder()
```

Comment:
I had three commit to handle this issue differently, and I think that lead to simpler code : https://github.com/linagora/james-project/pull/619/commits/a7fe3ec962ddf54826213c07520faf71589e5574 https://github.com/linagora/james-project/pull/619/commits/128d5a54054123fa44c70a871cc494ff30ede853 https://github.com/linagora/james-project/pull/619/commits/3d9bfae9afe5ef4dba8fdf2ad1e6646bc562cae8